src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -469,35 +469,53 @@ private BoundPattern BindDeconstructionPattern(DeconstructionPatternSyntax node,
-                    diagnostics.Add(ErrorCode.ERR_WrongNumberOfSubpatterns, location, declType.TupleElementTypes, elementTypes.Length, node.SubPatterns.Count);
+                    diagnostics.Add(ErrorCode.ERR_WrongNumberOfSubpatterns, location, declType, elementTypes.Length, node.SubPatterns.Count);
                     hasErrors = true;
                 }
                 for (int i = 0; i < node.SubPatterns.Count; i++)
                 {
+                    var subPattern = node.SubPatterns[i];
                     bool isError = i >= elementTypes.Length;
                     TypeSymbol elementType = isError ? CreateErrorType() : elementTypes[i];
-                    // PROTOTYPE(patterns2): Check that node.SubPatterns[i].NameColon?.Name corresponds to tuple element i of declType.
-                    BoundPattern boundSubpattern = BindPattern(node.SubPatterns[i].Pattern, elementType, isError, diagnostics);
+                    if (subPattern.NameColon != null)
+                    {
+                        string name = subPattern.NameColon.Name.Identifier.ValueText;
+                        FieldSymbol foundField = CheckIsTupleElement(subPattern.NameColon.Name, (NamedTypeSymbol)declType, name, i, diagnostics);
+                        // PROTOTYPE(patterns2): Should the tuple field binding for the name be stored somewhere in the node?
+
+                    }
+                    BoundPattern boundSubpattern = BindPattern(subPattern.Pattern, elementType, isError, diagnostics);
                     patterns.Add(boundSubpattern);
                 }
             }
             else
             {
                 // It is not a tuple type. Seek an appropriate Deconstruct method.
                 var inputPlaceholder = new BoundImplicitReceiver(node, declType); // A fake receiver expression to permit us to reuse binding logic
+                // PROTOTYPE(patterns2): include element names node.SubPatterns[i].NameColon?.Name in the AnalyzedArguments
+                // used in MakeDeconstructInvocationExpression so they are used to disambiguate.
                 BoundExpression deconstruct = MakeDeconstructInvocationExpression(
                     node.SubPatterns.Count, inputPlaceholder, node, diagnostics, outPlaceholders: out ImmutableArray<BoundDeconstructValuePlaceholder> outPlaceholders);
                 deconstructMethod = deconstruct.ExpressionSymbol as MethodSymbol;
-                // PROTOTYPE(patterns2): Set and check the deconstructMethod
-
+                int skippedExtensionParameters = deconstructMethod?.IsExtensionMethod == true ? 1 : 0;
                 for (int i = 0; i < node.SubPatterns.Count; i++)
                 {
+                    var subPattern = node.SubPatterns[i];
                     bool isError = outPlaceholders.IsDefaultOrEmpty || i >= outPlaceholders.Length;
```

Comment:
I think `MakeDeconstructInvocationExpression` will either return `default` for `outPlaceholders` or exactly the requested number.
In deconstruction binding, we check `HasAnyErrors` on the result from `MakeDeconstructInvocationExpression` instead.

Reply:
`HasAnyErrors` is too blunt a tool, as it then suppresses processing for *any* kind of nested error. I prefer this.

---
In reply to: [179892024](https://github.com/dotnet/roslyn/pull/25963#discussion_r179892024) [](ancestors = 179892024)